src/charmed_kubeflow_chisme/bundle/_bundle.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Helper to work with Juju bundles, preserving comments in the YAML"""
+
+import copy
+from pathlib import Path
+from typing import Optional
+
+from deepdiff import DeepDiff
+from ruamel.yaml import YAML
+
+from ..juju import Juju
+
+
+class Bundle:
+    """Juju bundle loader/dumper.
+
+    This class uses ruamel.yaml instead of the typical pyyaml in order to preserve all comments,
+    the order of the yaml, etc.  This way we can load/dump a yaml with comments without losing them
+    or reordering the items.
+    """
+
+    def __init__(self, filename: Optional[str] = None):
+        self._filename = filename
+        self._data = None
+        if self._filename:
+            self.load_bundle()
+
+    def deepcopy(self):
+        """Returns a new deep copy of this Bundle"""
+        newbundle = copy.deepcopy(self)
+        return newbundle
+
+    def diff(self, other):
+        """Returns a diff between this and another object, in DeepDiff format"""
+        return DeepDiff(self._data, other._data, ignore_order=True)
+
+    def dump(self, filename: str):
+        """Dumps as yaml to a file"""
+        with open(filename, "w") as fout:
+            yaml = YAML(typ="rt")
+            yaml.dump(self.to_dict(), fout)
+
+    def get_latest_revisions(self):
+        """Return a new Bundle that has the latest revisions of all charms in this bundle"""
+        newbundle = self.deepcopy()
+
+        applications = newbundle.applications
+
+        for name, application in applications.items():
+            # Skip charms we're not "tracking" a channel for
+            if "_tracked_channel" not in application:
+                continue
+
+            charm_name = application["charm"]
+            channel = application["_tracked_channel"]
+            revision = application.get("revision", None)
+
+            newest_revision = get_newest_charm_revision(charm_name, channel)
+
+            if revision != newest_revision:
+                # If revision is not up to date - update in situ from tracked channel
+                application["revision"] = newest_revision
+
+        return newbundle
+
+    def load_bundle(self):
+        """Loads a YAML file as a bundle"""
+        yaml = YAML(typ="rt")
+        self._data = yaml.load(Path(self._filename).read_text())
```

Comment:
This might fail quite easily

Reply:
Yeah fair point, although I'm not sure what to do to harden it.  Thoughts?

Reply:
Lets go with some descriptive exception ... I mean the problem would be if sb will use just this function (as its not preffixed with underscore so sb might just use it).

Reply:
Added a bit of description.  Does this help?